source/posix/host_resolver.c

```diff
@@ -41,13 +41,18 @@ int aws_default_dns_resolve(
     const char *hostname_cstr = aws_string_c_str(host_name);
     AWS_LOGF_DEBUG(AWS_LS_IO_DNS, "static: resolving host %s", hostname_cstr);
 
+    /* Android would prefer NO HINTS IF YOU DON'T MIND, SIR */
+#ifdef ANDROID
+    int err_code = getaddrinfo(hostname_cstr, NULL, NULL, &result);
+#else
     struct addrinfo hints;
     AWS_ZERO_STRUCT(hints);
     hints.ai_family = AF_UNSPEC;
     hints.ai_socktype = SOCK_STREAM;
     hints.ai_flags = AI_ALL | AI_V4MAPPED;
 
     int err_code = getaddrinfo(hostname_cstr, NULL, &hints, &result);
```

Comment:
Do we still getting a union of v4 and v6 in the results?

Reply:
If you're asking if we support it, yes we do.

Reply:
To clarify was just wondering it that's what we get back without hints.

Reply:
Yes. It boils down to what the interfaces available support, and it's a union between them.